### ISSUE TITLE
Please enter a valid security type

### ISSUE THREAD

**Issue by lopezfelipe93** [softwarespartan/IB4m#60]
Hello Abel, 
Thanks for the framework it is a great way for mac users to connect to IB since the trading toolbox is not enabled for mac. 

I'm getting this message error while working the Historical Data Example and I can't seem to work around it. Can you give me some insight please? 

session.eClientSocket.reqHistoricalData(1000001,contract,'20141230 16:00:00','1 Y','1 day','TRADES',1,1,false,[]); pause(0.5);

1000001 321 Error validating request:-'bP' : cause - Please enter a valid security type

Thanks a lot.

**Comment by softwarespartan** [comment]
Hi Felipe

Thanks for reaching out.

Are you able to share how the contract was configured?  This usually means that `contract.secType()` has not been set/configured properly.

Be sure to check out the IB API docs here:

https://interactivebrokers.github.io/tws-api/basic_contracts.html

For example:

```
contract.secType("STK");
contract.currency("USD");
contract.exchange("SMART");
contract.primaryExch("ISLAND");
```

**Comment by lopezfelipe93** [comment]
Thanks for replying, 

The issue is when working with ISLAND I get an error of no market permission.

162 Historical Market Data Service error message:No market data permissions for ISLAND STK

**Comment by softwarespartan** [comment]
Yep, that is a valid error.  This happens when your IB account doesn’t have the right permissions to access certain data.

See the docs here:

https://interactivebrokers.github.io/tws-api/market_data.html

Historical market data has same subscription requirements.

**Comment by lopezfelipe93** [comment]
Oh I see, I'm using a demo account while I get to know around the API so I guess I need a full account to work this examples?

Thanks a lot for your help

**Comment by ecpgieicg** [comment]
@lopezfelipe93 Since you have received a valid error from the server, you can get more info by reaching out to the IB customer service. ext 4 for api/software support. One of the things the API support _is_ able to do is to tell you what each error code means and where to look such things up.
